website/content/docs/operations/aws-oidc-provider.mdx

```diff
@@ -0,0 +1,361 @@
+      },
+    ]
+  })
+
+  tags = {
+    tag-key = "tag-value"
+  }
+}
+
+resource "aws_iam_policy" "policy" {
+  name        = "nomad-oidc-policy"
+  description = "A policy for federated Nomad OIDC"
+  policy      = data.aws_iam_policy_document.assume_role.json
+}
+
+resource "aws_iam_role_policy_attachment" "test-attach" {
+  role       = aws_iam_role.s3_all_access_role.name
+  policy_arn = aws_iam_policy.policy.arn
+}
+```
+
+This configuration requires the following information:
+- `<AWS_ACCOUNT_ID>`: The ID of the AWS account where the IAM role from the previous step was
+created.
+
+## Update the Nomad server configuration
+
+After you configure AWS, modify the Nomad server configuration file. Add the
+[`oidc_issuer` attribute](/nomad/docs/configuration/server#oidc_issuer) and set the value
+to the domain name for the Nomad cluster. This [enables the HTTP endpoint in Nomad](https://developer.hashicorp.com/nomad/docs/configuration/server#oidc_issuer) that allows third parties to discover Nomad's OIDC
+configuration.
+
+```hcl
+server {
+  enabled = true
+  [...]
+  oidc_issuer = "https://<DOMAIN_NAME>"
+  [...]
+}
+```
+
+This configuration requires the following information:
+- `<DOMAIN_NAME>`: The domain name of your Nomad cluster without the protocol or port.
+
+Restart the Nomad server agent to apply the configuration changes.
+
+## Create and run a sample jobspec file
+
+Create and run a jobspec file to validate your configuration. The following file is named `s3-upload.nomad.hcl`, add the following configuration to it, and
+save the file.
+
+```hcl
+job "s3" {
+  type = "batch"
+  group "bucket" {
+    task "copy" {
+      driver = "docker"
+      config {
+        image = "public.ecr.aws/aws-cli/aws-cli"
+        command = "s3"
+        args = ["cp", "/local/test.txt", "s3://<S3_BUCKET_NAME>/test-nomad.txt"]
+      }
+
+      identity {
+        name = "aws"
+        aud = ["aws"] 
+        file = true
+        env = true
+        ttl = "1h" 
+        change_mode = "restart"
+      }
+
+      template {
+        destination = "local/test.txt"
+        change_mode = "restart"
+        data        = <<EOF
+Job:          {{ env "NOMAD_JOB_NAME" }}
+Alloc:        {{ env "NOMAD_ALLOC_ID" }}
+EOF
+      }
+
+      env {
+        AWS_ROLE_ARN = "arn:aws:iam::<AWS_ACCOUNT_ID>:role/<IAM_ROLE_NAME>"
+        # The format of the token file is nomad_$NAME_OF_IDENTITY.jwt
+        AWS_WEB_IDENTITY_TOKEN_FILE = "${NOMAD_SECRETS_DIR}/nomad_aws.jwt"
+      }
+
+      resources {
+        cpu    = 500
+        memory = 256
+      }
+    }
+  }
+}
+```
+
+This configuration requires the following information:
+- `<S3_BUCKET_NAME>`: The name of the S3 bucket where the test file will be saved.
+- `<AWS_ACCOUNT_ID>`: The ID of the AWS account where the IAM role from the previous step was
+created.
+- `<IAM_ROLE_NAME>`: The name of the IAM role from previous steps. This will be
+`s3_all_access_role` if you are using the examples from this page.
```

Comment:
We should probably have these as [HCL2 Variables](https://developer.hashicorp.com/nomad/docs/job-specification/hcl2/variables)?

Reply:
Ideally yes, it's intended to be a usage guide and not a tutorial though so I wanted to keep the snippets minimal so it's easier to quickly parse